homeassistant/components/efesto/climate.py

```diff
@@ -0,0 +1,290 @@
+        return SUPPORT_TARGET_TEMPERATURE | SUPPORT_FAN_MODE
+
+    @property
+    def device_state_attributes(self):
+        """Return the device specific state attributes."""
+        return {
+            ATTR_DEVICE_STATUS: self._device_status,
+            ATTR_HUMAN_DEVICE_STATUS: self._human_device_status,
+            ATTR_SMOKE_TEMP: self._smoke_temperature,
+            ATTR_REAL_POWER: self._real_power,
+        }
+
+    @property
+    def unique_id(self):
+        """Return a unique ID."""
+        return self._device_id
+
+    @property
+    def name(self):
+        """Return the name of the Efesto, if any."""
+        return self._name
+
+    @property
+    def device_info(self):
+        """Return the device info."""
+        return {
+            "identifiers": {(DOMAIN, self.unique_id)},
+            "name": self.name,
+            "manufacturer": "Micronova",
+            "model": "Efesto",
+        }
+
+    @property
+    def precision(self):
+        """Return the precision of the system."""
+        return PRECISION_WHOLE
+
+    @property
+    def target_temperature_step(self):
+        """Return the supported step of target temperature."""
+        return PRECISION_WHOLE
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measurement."""
+        return TEMP_CELSIUS
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._current_temperature
+
+    @property
+    def target_temperature(self):
+        """Return the temperature we try to reach."""
+        return self._target_temperature
+
+    @property
+    def hvac_mode(self):
+        """Return hvac operation ie. heat, cool mode.
+
+        Need to be one of HVAC_MODE_*.
+        """
+        if self._on:
+            return HVAC_MODE_HEAT
+        return HVAC_MODE_OFF
+
+    @property
+    def hvac_modes(self):
+        """Return the list of available hvac operation modes.
+
+        Need to be a subset of HVAC_MODES.
+        """
+        return [HVAC_MODE_HEAT, HVAC_MODE_OFF]
+
+    @property
+    def fan_mode(self):
+        """Return fan mode."""
+        if self._current_power in FAN_MODES:
+            return self._current_power
+        return FAN_1
+
+    @property
+    def fan_modes(self):
+        """Return the list of available fan modes."""
+        return FAN_MODES
+
+    @property
+    def hvac_action(self):
+        """Return the current running hvac operation if supported.
+
+        Need to be one of CURRENT_HVAC_*.
+        """
+        if self._human_device_status in CURRENT_HVAC_MAP_EFESTO_HEAT:
+            return CURRENT_HVAC_MAP_EFESTO_HEAT.get(self._human_device_status)
+        return CURRENT_HVAC_IDLE
+
+    def turn_off(self):
+        """Turn device off."""
+        try:
+            self.client.set_off()
+        except EfestoError as err:
+            _LOGGER.error("Failed to turn off device (original message: %s)", err)
```

Comment:
Instead of wrapping the entire stack trace in braces:
```suggestion
            _LOGGER.error("Failed to turn off device. Error message: %s", err)
```